RCTComms/transport.py

```diff
@@ -401,3 +400,82 @@ def send(self, data: bytes, dest):
 
     def isOpen(self):
         return self.__socket is not None
+
+class RCTSerialTransport(RCTAbstractTransport):
+    '''
+    Serial Transport
+    socket.socket wrapped in a serial.Serial object wrapped in an abstract transport (move over, turducken)
+    '''
+    def __init__(self, port: str) -> None:
+        '''
+        Constructor for an RCTSerialTransport
+        :param port: port to be used in underlying socket connection
+        '''
+        self.__port = port
+        self.__serial: Optional[serial.Serial] = None
+
+    def open(self) -> None:
+        '''
+        Open the serial port.
+        '''
+        if self.__serial is None:
+            self.__serial = serial.Serial(self.__port)
+            self.__serial.set_buffer_size(rx_size=65536)
+        if not self.__serial.isOpen():
+            self.__serial.open()
+
+    def receive(self, buffer_len: int, timeout: int=None) -> Tuple[bytes, str]:
+        '''
+        Receive up to bufLen bytes of data from the port within timeout sec.
+
+        :param bufLen: Maximum number of bytes to return
+        :param timeout: Maximum number of seconds to wait for data
+
+        :return data, sender: Tuple containing the bytes received (data) and the
+                machine which sent that data (sender)
+        '''
+        if not self.__serial.isOpen():
```

Comment:
I don't think serial.Serial.isOpen exists: https://pythonhosted.org/pyserial/pyserial_api.html#native-ports

Reply:
Strange. Seems documentation might be outdated, as the function seems to exist when I tried testing: 
```
(rctGCS) >python
>>> import serial
>>> myserial = serial.Serial('COM1')
>>> myserial.isOpen()
True
>>> quit()
```